controllers/knative/knative.go

```diff
@@ -64,8 +81,23 @@ func NewKnativeEventingClient(cfg *rest.Config) (*clienteventingv1.EventingV1Cli
 	return clienteventingv1.NewForConfig(cfg)
 }
 
+func GetDisvoveryClient(cfg *rest.Config) (discovery.DiscoveryInterface, error) {
```

Comment:
```suggestion
func GetDiscoveryClient(cfg *rest.Config) (discovery.DiscoveryInterface, error) {
```

Reply:
I'm not sure if this is the right place for this function since it's not related directly to Knative. We have two options: either move it to the `utils/kubernetes` package or make it private if only Knative is using it. But IIRC, our service discovery use cases also use it.

Reply:
Changed to a local function